exercises/sam_jenkins/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * Skeleton main routine
+ */
+#include <string>
+#include <vector>
+#include <iostream>
+#include "Shapes.h"
+#include "Square.h"
+#include "Rectangle.h"
+#include "Circle.h"
+#include "Triangle.h"
+#include "ShapeSorter.h"
+using std::cout; using std::cin; using std::endl; using std::string; using std::vector;
+
+int main(int, char **)
+{
+	vector<Shapes*> shapes(10);
```

Comment:
I do not think we need to modify the shapes in the vector so they should be declared `vector<Shapes const *>`

Reply:
Also, here it would probably be nicer to use braced initilisation:

```cpp
vector<Shapes const *> shapes = { ... };
```